plugins/inputs/http_listener_v2/http_listener_v2.go

```diff
@@ -151,18 +158,68 @@ func (h *HTTPListenerV2) Init() error {
 		return err
 	}
 
+	protoRegex := regexp.MustCompile(`\w://`)
+	if !protoRegex.MatchString(h.ServiceAddress) {
+		h.ServiceAddress = "tcp://" + h.ServiceAddress
+	}
+
+	var u *url.URL
+	// Parse and check the address
+	if runtime.GOOS == "windows" && strings.HasPrefix(h.ServiceAddress, "unix://") {
+		u = &url.URL{Scheme: "unix", Path: strings.TrimPrefix(h.ServiceAddress, "unix://")}
+	} else {
+		u, err = url.Parse(h.ServiceAddress)
+		if err != nil {
+			return fmt.Errorf("parsing address failed: %w", err)
+		}
+	}
+
+	address := u.Host
+
+	switch u.Scheme {
+	case "tcp":
+	case "unix":
+		path := filepath.FromSlash(u.Path)
+		if runtime.GOOS == "windows" && strings.Contains(path, ":") {
+			path = strings.TrimPrefix(path, `\`)
+		}
+		if err := os.Remove(path); err != nil && !errors.Is(err, os.ErrNotExist) {
+			return fmt.Errorf("removing socket failed: %w", err)
+		}
+		address = path
+	default:
+		return fmt.Errorf("unknown protocol %q", u.Scheme)
+	}
+
 	var listener net.Listener
 	if tlsConf != nil {
-		listener, err = tls.Listen("tcp", h.ServiceAddress, tlsConf)
+		listener, err = tls.Listen(u.Scheme, address, tlsConf)
 	} else {
-		listener, err = net.Listen("tcp", h.ServiceAddress)
+		listener, err = net.Listen(u.Scheme, address)
 	}
 	if err != nil {
 		return err
 	}
 	h.tlsConf = tlsConf
 	h.listener = listener
-	h.Port = listener.Addr().(*net.TCPAddr).Port
+
+	if u.Scheme == "tcp" {
+		h.Port = listener.Addr().(*net.TCPAddr).Port
+	}
```

Comment:
Is the port used somewhere? If not, we should not set it and deprecate the config option!

Reply:
Looks like it isn't lets deprecate! - [dfbaf83](https://github.com/influxdata/telegraf/pull/15764/commits/dfbaf831e5757738a736018226361a58ba297094)